01DetectingFakeNews/01DetectingFakeNews.ipynb

```diff
@@ -0,0 +1,280 @@
+          ]
+        }
+      ],
+      "source": [
+        "from sklearn.model_selection import train_test_split\n",
+        "X_train, X_test, y_train, y_test = train_test_split(dataset['text'], labels, test_size = 0.2, random_state = 7)\n",
+        "print(y_train)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "SXfH5_AtdE00"
+      },
+      "source": [
+        "## Initialize TF-IDF-vectorizer"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "id": "BqIRCiJKdRcS"
+      },
+      "outputs": [],
+      "source": [
+        "from sklearn.feature_extraction.text import TfidfVectorizer\n",
+        "vectorizer = TfidfVectorizer(stop_words='english', max_df=0.7)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "e6ThG4UWdj-p"
+      },
+      "source": [
+        "## Fit and Transform training set. Transform test set."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 6,
+      "metadata": {
+        "id": "AdTderEldwWw"
+      },
+      "outputs": [],
+      "source": [
+        "v_train = vectorizer.fit_transform(X_train)\n",
+        "v_test = vectorizer.transform(X_test)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "hCJrVae9fb17"
+      },
+      "source": [
+        "## Initialize PassiveAgressiveClassifier"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 7,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "eA3Aoy0Mfg7X",
+        "outputId": "aabb2b0f-0d71-4b92-86ab-0c2893416391"
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "PassiveAggressiveClassifier(max_iter=50)"
+            ]
+          },
+          "execution_count": 7,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "from sklearn.linear_model import PassiveAggressiveClassifier\n",
+        "pac=PassiveAggressiveClassifier(max_iter=50)\n",
+        "pac.fit(v_train,y_train)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Uco2a7Kafs-8"
+      },
+      "source": [
+        "## Predict the test set results"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 8,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "-vyXJTXHfxd1",
+        "outputId": "dcf4a115-cc73-44ae-ad77-a8a4c5b79d25"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Accuracy: 92.27%\n"
+          ]
+        }
+      ],
+      "source": [
+        "from sklearn.metrics import accuracy_score\n",
```

Comment:
the accuracy is great but it can have some drawbacks (for example in case of imbalanced data, see https://developers.google.com/machine-learning/crash-course/classification/accuracy)
please calculate other metrics also, like precision, recall, f1-score https://neptune.ai/blog/performance-metrics-in-machine-learning-complete-guide